### ISSUE TITLE
[Button] Missing theming hooks

### ISSUE THREAD

**Issue by alex-golubtsov** [mineral-ui/mineral-ui#746]
As a consumer of Button component, I would like to have `Button_borderWidth_active` (since `Button_borderWidth` is now applied for `active` button state) and `Button_color_hover` to be able to change also text color on hover. In general I would like to have more control over Button theming.

**Comment by kylegach** [comment]
Hi Alex,

We won't have the bandwidth to work on this anytime soon. Would you be willing to submit a PR ([contribution guidelines](https://github.com/mineral-ui/mineral-ui/blob/master/CONTRIBUTING.md)) to add those theme variables and the associated styles?

To get you started, you'd add your theme variables here:

https://github.com/mineral-ui/mineral-ui/blob/master/src/library/Button/Button.js#L36-L72

Then write the styles here:

https://github.com/mineral-ui/mineral-ui/blob/master/src/library/Button/Button.js#L169-L199

Which will probably necessitate extending the `chooseColor` function, here:

https://github.com/mineral-ui/mineral-ui/blob/master/src/library/Button/Button.js#L74-L84

We're available to answer any questions that may come up. We can have those discussions either here, in the PR, or in some other channel like email, your choice.

Thanks,

Kyle

**Comment by G0L0L** [comment]
Hi @kylegach, I have a question

```
const Btn = createThemedComponent(Button, {
   Button_color_primary: '#ad5f00'
});
return <Btn variant="success" primary>Success</Btn>;
```

In this case `theme.color_SuccessPrimary` has higher priority, then `Button_color_primary` I defined in `createThemedComponent`. Is it expected behavior?

**Comment by kylegach** [comment]
Yes, that is expected. The `danger`, `success`, and `warning` variants are cross-component concepts, so we do not expose component-level theme variables to adjust them. If you really need to override them, you can do so by referencing the "global" variable, rather than the Button one:

```
const Btn = createThemedComponent(Button, {
  color_successPrimary: '#ad5f00'
});
```

But before you do that, consider placing that override in your root ThemeProvider, so that it applies to all components, not just Btn.
